variables.tf

```diff
@@ -114,6 +114,25 @@ variable "storage_disk_size" {
   }
 }
 
+variable "additional_worker_groups" {
+  type    = map(object({ size = string, count = number, disk_size = number, state = string }))
+  default = {}
+
+  validation {
+    condition = alltrue([
+      for k, v in var.additional_worker_groups :
+      !contains(["worker", "master", "infra", "storage"], k) &&
+      v.count > 0 &&
+      v.disk_size >= 120 &&
+      (v.state == "Running" || v.state == "Stopped")
+    ])
+    // Cannot use any of the nicer string formatting options because
+    // error_message validation is dumb, cf.
+    // https://github.com/hashicorp/terraform/issues/24123
+    error_message = "Your configuration of `additional_worker_groups` violates one of the following constraints:\n * The minimum supported disk size for workers is 120GB.\n * Additional worker group names cannot be 'worker', 'master', 'infra', or 'storage'.\n * The only valid worker states are 'Running' or 'Stopped'.\n * The worker count cannot be less than 0."
```

Comment:
The error message ends up looking as follows:

```
$ terraform validate

Error: Invalid value for variable

  on main.tf.json line 16, in module.cluster:
  16:       "additional_worker_groups": {
  17:         "test": {
  18:           "size": "Extra-large",
  19:           "count": -1,
  20:           "disk_size": 128,
  21:           "state": "Running"
  22:         }
  23:       }

Your configuration of `additional_worker_groups` violates one of the following
constraints:
 * The minimum supported disk size for workers is 120GB.
 * Additional worker group names cannot be 'worker', 'master', 'infra', or 'storage'.
 * The only valid worker states are 'Running' or 'Stopped'.
 * The worker count cannot be less than 0.

This was checked by the validation rule at
.terraform/modules/cluster/variables.tf:121,3-13.

```

Reply:
This actually also has an upside. When I violate several rules, I get a reminder for everything that is checked. Usually, I get just the first one and then the second and so on.